bin/pipeline_result_processor/prp/models/phenotype.py

```diff
@@ -0,0 +1,82 @@
+"""Datamodels used for prediction results."""
+from enum import Enum
+from typing import Dict, List
+
+from pydantic import BaseModel, Field
+
+from .base import RWModel
+
+
+class VariantType(Enum):
+    SUBSTITUTION = "substitution"
+
+
+class PhenotypeType(Enum):
+    AMR = "antimicrobial_resistance"
+    CHEM = "chemical_resistance"
+    ENV = "environmental_factor_resistance"
+    VIR = "virulence"
+
+
+class DatabaseReference(RWModel):
+    ref_database: str
+    ref_id: str
+
+
+class GeneBase(BaseModel):
+    """Container for gene information"""
+
+    name: str
+    accession: str
+    # prediction info
+    depth: float | None
+    identity: float
+    coverage: float
+    ref_start_pos: int
+    ref_end_pos: int
+    ref_gene_length: int
+    alignment_length: int
+
+
+class ResistanceGene(GeneBase, DatabaseReference):
+    """Container for resistance gene information"""
+
+    phenotypes: List[str]
+
+
+class VirulenceGene(GeneBase, DatabaseReference):
+    """Container for virulence gene information"""
+
+    virulence_category: str
```

Comment:
Can be a bit confusing that VirulenceGene and ResistanceGene contains different data types and effectively completely different types of data. A suble name change can be useful